Proteomics/ProteolyticDigestion/PeptideWithSetModifications.cs

```diff
@@ -260,15 +267,15 @@ public CompactPeptide CompactPeptide(TerminusType terminusType)
 
         public PeptideWithSetModifications Localize(int j, double massToLocalize)
         {
-            var vvv = new Dictionary<int, ModificationWithMass>(AllModsOneIsNterminus);
+            var vvv = new Dictionary<int, ModificationGeneral>(AllModsOneIsNterminus);
             double massOfExistingMod = 0;
-            if (vvv.TryGetValue(j + 2, out ModificationWithMass modToReplace))
+            if (vvv.TryGetValue(j + 2, out ModificationGeneral modToReplace))
             {
-                massOfExistingMod = modToReplace.monoisotopicMass;
+                massOfExistingMod = (double)modToReplace.MonoisotopicMass;
                 vvv.Remove(j + 2);
             }
 
-            vvv.Add(j + 2, new ModificationWithMass(null, null, null, TerminusLocalization.Any, massToLocalize + massOfExistingMod));
+            vvv.Add(j + 2, new ModificationGeneral(null, null, null, null, null, "Anywhere.", null, massToLocalize + massOfExistingMod, null, null, null, null, null, null));
```

Comment:
clean this constructor up with the parameter-specifying syntax. i.e. `new Modification(MonoisotopicMass: massToLocalize + massOfExistingMod, ....... );`